Parser/asdl_c.py

```diff
@@ -1132,13 +1132,93 @@ def visitModule(self, mod):
     return result;
 }
 
+/* copy.replace() support (shallow copy) */
+static PyObject *
+ast_type_replace(PyObject *self, PyObject *args, PyObject *kwargs) {
+    if (!_PyArg_NoPositional("__replace__", args)) {
+        return NULL;
+    }
+
+    struct ast_state *state = get_ast_state();
+    if (state == NULL) {
+        return NULL;
+    }
+
+    PyObject *result = NULL;
+    // known AST class fields
+    PyObject *fields = NULL;
+    // current instance dictionary
+    PyObject *__dict__ = NULL;
+    // constructor positional and keyword arguments
+    PyObject *empty_tuple = NULL, *payload = NULL;
+    // for iterating over 'fields' and 'kwargs'
+    PyObject *key = NULL, *value = NULL;
+
+    PyTypeObject *type = Py_TYPE(self);
+    if (PyObject_GetOptionalAttr((PyObject *)type, state->_fields, &fields) < 0) {
+        goto cleanup;
+    }
+    if (PyObject_GetOptionalAttr(self, state->__dict__, &__dict__) < 0) {
+        goto cleanup;
+    }
+    if (!(empty_tuple = PyTuple_New(0))) {
+        goto cleanup;
+    }
+    if (!(payload = PyDict_New())) {
+        goto cleanup;
+    }
+    if (fields) {
+        Py_ssize_t numfields = PySequence_Size(fields);
+        if (numfields == -1) {
+            goto cleanup;
+        }
+        for (Py_ssize_t i = 0; i < numfields; i++) {
+            if (!(key = PySequence_GetItem(fields, i))) {
```

Comment:
Do not use assignment in expression without need.
```suggestion
            key = PySequence_GetItem(fields, i);
            if (key == NULL) {
```
This is only justified in the `while` or `for` condition, in the `else if` condition, and maybe in few other cases.

Reply:
I assume that 
```C
  if (!(empty_tuple = PyTuple_New(0))) {
      goto cleanup;
  }
```

should also be rewritten as 

```C
empty_tuple = PyTuple_New(0);
if (empty_tuple == NULL) {
    goto cleanup;
}
```